cmake/FindRust.cmake

```diff
@@ -218,14 +218,13 @@ if (NOT "${Rust_TOOLCHAIN}" STREQUAL "$CACHE{Rust_TOOLCHAIN}")
     set(Rust_TOOLCHAIN ${Rust_TOOLCHAIN} CACHE STRING "Requested rustup toolchain" FORCE)
 endif()
 
+set(Rust_RESOLVE_RUSTUP_TOOLCHAINS CACHE BOOL ON)
```

Comment:
Yea, it works without a docstring. I can add a docstring but I wasn't sure how this behaves with later invocations of `set(... CAHCE)` (does it update the docstring? does the docstring remain as in the first `set` command?)

Reply:
> Yea, it works without a docstring.

Adding `message(STATUS "Rust_RESOLVE_RUSTUP_TOOLCHAINS is set to ${Rust_RESOLVE_RUSTUP_TOOLCHAINS}")` after this line prints:
```
-- Rust_RESOLVE_RUSTUP_TOOLCHAINS is set to 
```

The `CMakeCache.txt` contains:
```
//ON
Rust_RESOLVE_RUSTUP_TOOLCHAINS:BOOL=
```

Please put the value after the variable name as per the documented signature. The docstring is not optional, and should be the same for all instances. I would assume the bahviour for the docstring is the same as for the value. If the cache variable already exists, then it is not updated unless `FORCE` is specified.

Reply:
Oh, I see, it works because I don't use the value at all, only the existence of the flag

Reply:
I fixed it. But that's so weird that it's like that; how are you supposed to *just set* a CMake cache variable then? like, without resetting its description

Reply:
Alright, it seems like the expected use is to declare a CACHE variable once and to override its behavior by `set()`ing a regular variable, which, according to CMake laws overrides a CACHE variable in the evaluation order. *However*, in order for this to also work with user-defined non-cache variables (which are set before the CACHE variable is declared), we have to use [the CMP0126 policy](https://cmake.org/cmake/help/latest/policy/CMP0126.html).

Can you check out the last commit and tell me if you wanna merge it like this or without it?

Reply:
No,  the expected way would be to overwrite the current value of the cache variable by specifying FORCE. 

Setting a normal variable would confuse users, since the variable is only in scope for the duration of FindRust, and would confuse users if it is different than what they see in the cmake cache.

Reply:
Are you sure? It's very weird that it's like that. It means that if a user wants to set this option from their code, they have to also set the cache description to something new, for a variable that's not theirs.

Anyway I removed it since it seems like the GH Actions CMake is too old to have that policy

Reply:
> It means that if a user wants to set this option from their code, they have to also set the cache description to something new, for a variable that's not theirs.

Good point, I think for the other cache variables we check if the user defined a variable (normal or cache) with such a name, and then use that to initialize the cache variable, and fall-back to our default if the user did not define such a variable.

Reply:
I think that's exactly what CMP0126 attempts to remediate, though. Enabling it (on CMake 3.21 and above) makes it unnecessary to check if the user defined their own variable, because even if they did, that variable would remain and gain precedence over the CACHE variable in condition evaluations.